tests/depositWithdrawTest.ts

```diff
@@ -0,0 +1,263 @@
+import BN from "bn.js";
+import sha256 from "crypto-js/sha256";
+import hexEnc from "crypto-js/enc-hex";
+import { Field } from "delphinus-curves/src/field";
+import { withL1Client, L1Client } from "../clients/client";
+import { getConfigByChainName } from "delphinus-deployment/src/config";
+import { L1ClientRole } from "delphinus-deployment/src/types";
+import { encodeL1address } from "web3subscriber/src/addresses";
+import { PromiseBinder } from "web3subscriber/src/pbinder";
+
+
+async function mintToken(testChain: string) {
+    let config = await getConfigByChainName(L1ClientRole.Monitor, testChain);
+    let account = config.monitorAccount;
+    let pbinder = new PromiseBinder();
+    let r = pbinder.return(async () => {
+      await withL1Client(config, false, async (l1client: L1Client) => {
+        let token = l1client.getTokenContract();
+        try {
+          pbinder.snapshot("Mint");
+          console.log("mint token:", token.address());
+          let balance = await token.balanceOf(account);
+          if(balance < new BN(100)){
+            console.log("Monitor Account's balance before mint:", balance.toString(10));
+            await pbinder.bind("mint", token.mint(new BN("1000")));
+            balance = await token.balanceOf(account);
+            console.log("Monitor Account's balance:", balance.toString(10));
+          }else{
+            console.log("Monitor Account's balance:", balance.toString(10));
+            console.log("Monitor Account Have Enough Test Token To DO DEPOSIT & WITHDRAW TEST, SKIPED MINT");
+          }
+        } catch (err) {
+          console.log("%s", err);
+        }
+      });
+    });
+    await r.when(
+      "mint",
+      "transactionHash",
+      (hash: string) => console.log(hash)
+    );
+  }
+
+async function addToken(testChain: string){
```

Comment:
This one is not good.
For testing purpose PROXY contract, we can do this.
However for production, we cannot "adding a token to the production PROXY contract when we run test"
Thus better find a way to do "try to find the index of the TOKEN in the PROXY contract" and then use th index.
I think we probably can do this as we can get the details of tokens from the proxy (that status.ts had done that) and then compare the address with TOKEN address and find the index. (If multiple, just use the first one?)

Reply:
Ummm, anyway this is not that important as we have to use test purpose PROXY contract to do the testing as we send mock data to the PROXY.

So this scenario testing should to be done on testing PROXY contract only.

So we do not need change this one.

However, I think we need remind tests runner about this.
So I think we need write a reminder in the actionTest.sh and ask the user that PROXY contract address in PROXY.json is test purpose only and not production address, and verifier is Dummy. Until user press "Y", then we begin the testing.

Reply:
this function will check and add test token only if there isn't one, no other tokens.